src/main/java/DeadlineCommand.java

```diff
@@ -0,0 +1,53 @@
+import java.time.LocalDateTime;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DeadlineCommand extends Command {
+
+    public static final String COMMAND = "deadline";
+    private static final Pattern COMMAND_FORMAT = Pattern.compile("(.*)\\W(?=/by)/by\\W(.*)");
+    private static final String ERROR_MESSAGE =
+            "☹ Sorry, please enter a valid description and datetime for the deadline\n\t" +
+            "Command: deadline [description] /by [deadline]";
+    private final String description;
+    private final LocalDateTime by;
+
+    private DeadlineCommand(String description, LocalDateTime by) {
+        this.description = description;
+        this.by = by;
+    }
+
+    @Override
+    public String execute() throws DukeException {
+        Task newTask = new Deadline(description, by);
+        taskList.addTask(newTask);
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Got it. I've added this task:\n\t").append(newTask).append("\nNow you have ")
```

Comment:
It might assist readability to break the `.append` into multiple lines since they are new lines, and it makes it clear what is being appended to the `StringBuilder`.

This specific style issue exists in a few other places too.